infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -370,6 +370,29 @@ private void onLeadershipAcquire() {
         }
     }
 
+    /**
```

Comment:
One more thought is when we process config change event notification, we are using the config from that notification message, which might be delayed/outdated. I was wondering if we can fetch latest one to skip those outdated ones?

Reply:
If there is another change, it should send another notification. We should stick to one model, either push(get notified) or poll(query). If we sometime use "notification", sometime use "query", it complicate the system.

Reply:
mmm but that would be really complex right, the same argument could be thought for the one being fetched (that it could be delayed/outdated). Anyways, if it is delayed or outdated we'll soon receive another notification.

Reply:
While processing a topology event, it is an optimization we can do to query the most recent topology. But while there is a leadership change, there is no need to query the topology to complicate the scenario.